### PR TITLE
Refactor `bee-packable-derive`.

### PR DESCRIPTION
**Pull request by pvdrz**
This PR should make the crate a bit easier to maintain and update.

### CODE REVIEW

File: bee-common/bee-packable-derive/src/lib.rs [hunk 1]
```diff
@@ -23,7 +30,7 @@ use syn::parse_macro_input;
 #[proc_macro_error]
 #[proc_macro_derive(Packable, attributes(packable))]
 pub fn packable(input: TokenStream) -> TokenStream {
-    let input = parse_macro_input!(input);
+    let input: DeriveInput = parse_macro_input!(input);
```
Comment (thibault-martinez) [inline bee-common/bee-packable-derive/src/lib.rs]: Just curious, why is this needed ?

Comment (pvdrz) [reply bee-common/bee-packable-derive/src/lib.rs]: I think it is not, is just debris from my refactor.

File: bee-common/bee-packable-derive/src/enum_info.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use quote::quote;
+use syn::{parse2, Attribute, DataEnum, Ident, Result, Type};
+
+use crate::{
```
Comment (thibault-martinez) [inline bee-common/bee-packable-derive/src/enum_info.rs:4]: Crate before 3rd parties

File: bee-common/bee-packable-derive/src/record_info.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use syn::{Expr, Fields, Ident, Path, Result, Type};
+
+use crate::field_info::{FieldInfo, IdentOrIndex};
```
Comment (thibault-martinez) [inline bee-common/bee-packable-derive/src/record_info.rs:4]: Crate before 3rd parties

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

**Review by thibault-martinez — APPROVED** [review]
